launchlib/cgroup.go

```diff
@@ -0,0 +1,111 @@
+package launchlib
+
+import (
+	"bufio"
+	"bytes"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	selfCGroup    = "/proc/self/cgroup"
+	selfMountinfo = "/proc/self/mountinfo"
+)
+
+type CGroupName string
+
+type CGroupPather interface {
+	Path(name CGroupName) (string, error)
```

Comment:
Could see an argument for pushing the "value" reading here as well, so all of the io/fs path normalization stuff happens internally.